sklearn/inspection/_plot/decision_boundary.py

```diff
@@ -0,0 +1,318 @@
+            string is used.
+
+        ax : Matplotlib axes, default=None
+            Axes object to plot on. If `None`, a new figure and axes is
+            created.
+
+        **kwargs : dict
+            Additional keyword arguments to be passed to the
+            `plot_method`.
+
+        Returns
+        -------
+        display : :class:`~sklearn.inspection.DecisionBoundaryDisplay`
+            Object that stores the result.
+
+        See Also
+        --------
+        DecisionBoundaryDisplay : Decision boundary visualization.
+        ConfusionMatrixDisplay.from_estimator : Plot the confusion matrix
+            given an estimator, the data, and the label.
+        ConfusionMatrixDisplay.from_predictions : Plot the confusion matrix
+            given the true and predicted labels.
+
+        Examples
+        --------
+        >>> import matplotlib.pyplot as plt
+        >>> from sklearn.datasets import load_iris
+        >>> from sklearn.linear_model import LogisticRegression
+        >>> from sklearn.inspection import DecisionBoundaryDisplay
+        >>> iris = load_iris()
+        >>> X = iris.data[:, :2]
+        >>> classifier = LogisticRegression().fit(X, iris.target)
+        >>> disp = DecisionBoundaryDisplay.from_estimator(
+        ...     classifier, X, response_method="predict",
+        ...     xlabel=iris.feature_names[0], ylabel=iris.feature_names[1],
+        ...     alpha=0.5,
+        ... )
+        >>> disp.ax_.scatter(X[:, 0], X[:, 1], c=iris.target, edgecolor="k")
+        <...>
+        >>> plt.show()
+        """
+        check_matplotlib_support(f"{cls.__name__}.from_estimator")
+
+        if not grid_resolution > 1:
+            raise ValueError("grid_resolution must be greater than 1")
+
+        if not eps >= 0:
+            raise ValueError("eps must be greater than or equal to 0")
+
+        possible_plot_method = ("contourf", "contour", "pcolormesh")
+        if plot_method not in possible_plot_method:
+            raise ValueError(
+                f"plot_method must be one of {', '.join(possible_plot_method)}. "
+                f"Got {plot_method} instead."
+            )
+
+        x0, x1 = _safe_indexing(X, 0, axis=1), _safe_indexing(X, 1, axis=1)
+
+        x0_min, x0_max = x0.min() - eps, x0.max() + eps
+        x1_min, x1_max = x1.min() - eps, x1.max() + eps
+
+        xx0, xx1 = np.meshgrid(
+            np.linspace(x0_min, x0_max, grid_resolution),
+            np.linspace(x1_min, x1_max, grid_resolution),
+        )
+
+        pred_func = _check_boundary_response_method(estimator, response_method)
+        response = pred_func(np.c_[xx0.ravel(), xx1.ravel()])
+
+        if response.ndim != 1:
+            if response.shape[1] != 2:
+                raise ValueError(
+                    "Multiclass classifiers are only supported when "
+                    "response_method='predict'"
+                )
+            response = response[:, 1]
+
```

Comment:
I think that if `response.ndim == 1` and `pred_func.__name__ == "predict"`and `hasattr(estimator, "classes_")` we should use the latter to re-encode `response` using `classes_` to get integers starting at 0. and use the class labels in the plot with a legend.